rollup.config.js

```diff
@@ -0,0 +1,75 @@
+import typescript from 'rollup-plugin-typescript';
+import resolve from 'rollup-plugin-node-resolve';
+import commonjs from 'rollup-plugin-commonjs';
+import replace from 'rollup-plugin-replace';
+import { uglify } from 'rollup-plugin-uglify';
+
+const bundle = env => {
+  const file =
+    env === 'production'
+      ? 'dist/general-store.min.js'
+      : 'dist/general-store.js';
+  const additionalPlugins = [];
+  if (env === 'production') {
+    additionalPlugins.push(uglify());
+  }
+  const config = {
+    input: './src/GeneralStore.ts',
+    plugins: [
+      typescript(),
+      commonjs(),
+      replace({
+        'process.env.NODE_ENV': JSON.stringify(env),
+      }),
+      resolve({
+        browser: true,
+      }),
+      ...additionalPlugins,
+    ],
+    output: [
+      {
+        file,
+        format: 'umd',
+        exports: 'named',
+        compact: env === 'production',
+        globals: {
+          react: 'React',
+        },
+      },
+    ],
+    external: ['react'],
+  };
+  return config;
+};
+const moduleBundle = env => {
+  return {
+    input: './src/GeneralStore.ts',
+    plugins: [
+      typescript(),
+      commonjs(),
+      resolve({
+        browser: true,
+      }),
+      replace({
+        'process.env.NODE_ENV': JSON.stringify(env),
```

Comment:
i think we can remove `replace()` usage for the module bundles.

Reply:
any maybe rename `env` to `format` in this function?

Reply:
oops. yeah. copypasta

Reply:
actually don't need the globals field either, that's only for the UMD build

Reply:
I wonder if we want to use `process.env.NODE_ENV` checks to make `enforceDispatcher` and/or `enforceValidDependencies` a noop in production? Is it realistic that it would only happen in prod?